File: lecture_007(Strings)/l_004_printpalindromicsubstrings.java
import java.util.Scanner;
public class l_004_printpalindromicsubstrings {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        String s = scn.nextLine();
        printPalindromeSubstrings(s);
    }
    public static void display(String s) {
        for(int i=0; i<s.length(); i++) {
            System.out.print(s.charAt(i));
        }
        System.out.println();
    }
    public static boolean isPalindrome(String s) {
        int si = 0;
        int ei = s.length()-1;
        while(si<ei) {
            if(s.charAt(si) != s.charAt(ei)) {
                return false;
            }
            si++;
            ei--;
        }
        return true;
    }
    public static void printPalindromeSubstrings(String s) {
        for(int i=0; i<s.length(); i++) {
            for(int j=i+1; j<=s.length(); j++) {
                String v = s.substring(i,j);
                //check whether this substring is palindrome or not
                boolean palindromic = isPalindrome(v);
                if(palindromic==true) {
                    System.out.println(v);
                }
            }
        }
    }
}

File: lecture_007(Strings)/l_004_printpalindromicsubstrings.cpp
#include<iostream>
#include<string>
using namespace std;

void display(string s) {
    for(int i=0; i<s.size(); i++) {
        cout<<s[i]+" ";
    }
    cout<<endl;
}
bool ispalindrome(string s) {
    int si = 0;
    int ei = s.size()-1;
    while(si<ei) {
        if(s[si] != s[ei]) {
            return false;
        }
        si++;
        ei--;
    }
    return true;
}
void printpalindromicsubstrings(string s) {
    for(int i=0; i<s.size(); i++) {
        for(int j=1; j<=s.size()-i; j++) {
            string v = s.substr(i,j);
            //check whether this substring is palindrome or not
            bool palindromic = ispalindrome(v);
            if(palindromic==true) {
                cout<<v<<endl;
            }
        }
    }
}
int main() {
    string s;
    cin>>s;
    //cout<<ispalindrome(s)<<endl;
    printpalindromicsubstrings(s);

    return 0;
}